src/main/java/no/nav/pensjon/selvbetjeningopptjening/opptjening/api/OpptjeningOnBehalfController.kt

```diff
@@ -0,0 +1,45 @@
+package no.nav.pensjon.selvbetjeningopptjening.opptjening.api
+
+import jakarta.servlet.http.HttpServletRequest
+import mu.KotlinLogging
+import no.nav.pensjon.selvbetjeningopptjening.consumer.FailedCallingExternalServiceException
+import no.nav.pensjon.selvbetjeningopptjening.opptjening.OpptjeningProvider
+import no.nav.pensjon.selvbetjeningopptjening.opptjening.Pid
+import no.nav.pensjon.selvbetjeningopptjening.opptjening.PidValidationException
+import no.nav.pensjon.selvbetjeningopptjening.opptjening.dto.OpptjeningResponse
+import no.nav.pensjon.selvbetjeningopptjening.tech.security.ingress.impersonal.audit.Auditor
+import no.nav.pensjon.selvbetjeningopptjening.tech.security.masking.Masker
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+
+@Deprecated("not allowed to use PID as request parameter")
+@RestController
+@RequestMapping("api")
+class OpptjeningOnBehalfController(
+    private val provider: OpptjeningProvider,
+    private val auditor: Auditor
+) {
+    private val log = KotlinLogging.logger { }
+
+    @GetMapping("/opptjeningonbehalf")
+    fun getOpptjeningOnBehalfOf(@RequestParam(value = "pid") pidValue: String?, request: HttpServletRequest): OpptjeningResponse? {
```

Comment:
Var det `requestParam` før også eller kommer den kryptert?

Reply:
Ja, var `requestParam` før også, dette er funksjonelt uendret.
PID er dessverre ikke kryptert, og det må fikses. Har opprettet egen sak for det: [TPP-20](https://jira.adeo.no/browse/TPP-20).